common/remote/sockfile.cpp

```diff
@@ -4020,20 +4171,216 @@ IRemoteActivity *createRemoteDiskRead(IPropertyTree &actNode)
+    virtual const void *nextRow(MemoryBufferBuilder &outBuilder, size32_t &retSz) override
+    {
+        if (eofSeen)
+            return nullptr;
+        checkOpen();
+        if (!eofSeen)
+        {
+            if (processed < chooseN)
+            {
+                while (keyManager->lookup(true))
+                {
+                    seeks += keyManager->querySeeks();
+                    scans += keyManager->queryScans();
+
+                    // JCSMORE -> GH - will need IVirtualFieldCallback blob field callback
+                    const byte *keyRow = keyManager->queryKeyBuffer();
+                    retSz = translator->translate(outBuilder, *this, keyRow);
+                    if (retSz)
+                    {
+                        const void *ret = outBuilder.getSelf();
+                        outBuilder.finishRow(retSz);
+                        ++processed;
+                        return ret;
+                    }
+                }
+                retSz = 0;
+            }
+            eofSeen = true;
+        }
+        close();
+        return nullptr;
+    }
+    virtual void serializeCursor(MemoryBuffer &tgt) const override
+    {
+        keyManager->serializeCursorPos(tgt);
+        tgt.append(processed);
+    }
+    virtual void restoreCursor(MemoryBuffer &src) override
+    {
+        checkOpen();
+        eofSeen = false;
+        keyManager->deserializeCursorPos(src);
+        src.read(processed);
+    }
+    virtual StringBuffer &getInfoStr(StringBuffer &out) const override
+    {
+        return out.appendf("indexread[%s]", helper->getFileName());
+    }
+//interface IVirtualFieldCallback
+    virtual unsigned __int64 getFilePosition(const void * row) override
+    {
+        return 0; // JCSMORE
+    }
+    virtual unsigned __int64 getLocalFilePosition(const void * row) override
+    {
+        return 0; // JCSMORE
+    }
+};
+
+
+IRemoteActivity *createRemoteIndexRead(IPropertyTree &actNode)
+{
+    const char *fileName = actNode.queryProp("fileName");
+    unsigned __int64 chooseN = actNode.getPropInt64("chooseN", defaultFileStreamChooseNLimit);
+    unsigned __int64 skipN = actNode.getPropInt64("skipN", defaultFileStreamSkipN);
+    bool isTlk = actNode.getPropBool("isTlk");
+    bool allowPreload = actNode.getPropBool("allowPreload");
+
+    Owned<IOutputMetaData> inMeta = getTypeInfoOutputMetaData(actNode, "input", false);
+    Owned<IOutputMetaData> outMeta = getTypeInfoOutputMetaData(actNode, "output", false);
+    if (!outMeta)
+        outMeta.set(inMeta);
+    Owned<IHThorIndexReadArg> helper = createIndexReadArg(fileName, LINK(inMeta), LINK(outMeta), LINK(outMeta), chooseN, skipN, (unsigned __int64)-1);
+    assertex(!helper->needTransform()); // Code assumes that we do not need to call the transform
+    IDynamicIndexReadArg *dynArg = QUERYINTERFACE(helper.get(), IDynamicIndexReadArg);
+    assertex(dynArg);
+    Owned<IPropertyTreeIterator> filterIter = actNode.getElements("keyFilter");
```

Comment:
minor: Slightly cleaner if this code was inside createIndexReadArg() as it is for createDiskReadArg()